db-engine/src/main/java/com/splicemachine/db/impl/sql/execute/TriggerExecutionContext.java

```diff
@@ -495,4 +518,49 @@ private static ExecRow extractColumns(ExecRow resultSet, boolean firstHalf) thro
         }
         return result;
     }
+
+    /**
+     * If the update trigger references a column, execRow only contains columns that are changed and referenced by
+     * trigger action. Extract columns according to heapList and triggerDescriptor
+     * @param resultSet
+     * @param firstHalf
+     * @return
+     * @throws SQLException
+     */
+    private ExecRow extractTriggerColumns(ExecRow resultSet, boolean firstHalf) throws SQLException {
+        int[] referencedCols = triggerd.getReferencedCols();
+        int[] referencedColsInTriggerAction = triggerd.getReferencedColsInTriggerAction();
+        int nResultCols = (referencedCols!=null?referencedCols.length:0) +
+                (referencedColsInTriggerAction!=null?referencedColsInTriggerAction.length:0);
+        ExecRow result = new ValueRow(nResultCols);
+        try {
+            FormatableBitSet allTriggerColumns = new FormatableBitSet(triggerd.getTableDescriptor().getNumberOfColumns()+1);
+            if (referencedCols != null) {
+                for (int col : referencedCols) {
+                    allTriggerColumns.set(col);
+                }
+            }
+            if (referencedColsInTriggerAction != null) {
+                for (int col : referencedColsInTriggerAction) {
+                    allTriggerColumns.set(col);
+                }
+            }
+            int nCols = resultSet.nColumns()/2;
+            int sourceIndex = (firstHalf ? 1 : nCols + 1);
+            int pos = 1;
+
+            for (int i = allTriggerColumns.anySetBit(); i >= 0; i = allTriggerColumns.anySetBit(i)) {
+                int index = 0;
+                for (int j = heapList.anySetBit(); j >= 0; j = heapList.anySetBit(j)) {
+                    if (j == i) {
```

Comment:
Minor comment.. this could use heapList.isSet(i) instead of looping through heapList.  Most likely no one will have a trigger with hundreds or columns or be updating hundreds of columns, so it's probably OK as-is.